### PR TITLE
update quickstart for IV first stage, diagnostics, Wald tests

### PR DESCRIPTION
**Pull request by s3alfisc**
- Updates the quickstart notebooks.
- Adds a section on IV diagnostics and first stage. 
- Adds a section on wald tests.

### CODE REVIEW

File: docs/quickstart.ipynb [hunk 1]
```diff
@@ -34,14 +34,14 @@
   },
```
Comment (Jayhyung) [inline docs/quickstart.ipynb:34]: <p>I think we can also add one more example of Wald test to make the user understand how to construct <code>R</code> matrix and <code>q</code>  vector like the following :</p><p> </p><p>You can also test the complicated joint null hypothesis of \X_1 + 2*X2 = 2.0\ and \X_2 = 1.0 \  by setting <code>R</code> and <code>q</code> as the following </p><p> </p>

<pre class="ql-syntax" spellcheck="false">R1 =np.array([[1,2], [0,1]])
q1 = np.array([2.0, 1.0])
fit.wald_test(R = R1, q = q1)
</pre><p> </p><p>How do you think?</p><p>	    </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/py-econometrics/pyfixest/pull/569/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='73'/>

[the remark this answers is not in the document]
Comment (s3alfisc) [reply docs/quickstart.ipynb:34]: Generally think it is good to have it - I think we might be able to catch and ignore it when calling the wald test on the Feols method. I do something like it in the `Fepois` class [here](https://github.com/py-econometrics/pyfixest/blob/5e1bd106448bb87bd70906aca65e5c719440cdc4/pyfixest/estimation/fepois_.py#L162): 

```py
        def compute_deviance(_Y: np.ndarray, mu: np.ndarray):
            with warnings.catch_warnings():
                warnings.simplefilter("ignore")
                deviance = (
                    2 * np.sum(np.where(_Y == 0, 0, _Y * np.log(_Y / mu)) - (_Y - mu))
                ).flatten()
            return deviance
```

[the remark this answers is not in the document]
Comment (s3alfisc) [reply docs/quickstart.ipynb:34]: I'll add the filter in the next commit =)

Comment (s3alfisc) [reply docs/quickstart.ipynb:34]: Solved with [a86f14d](https://github.com/py-econometrics/pyfixest/pull/569/commits/a86f14d98c426c45b09c901f66ec07dd8c7b36d1) =)

[the remark this answers is not in the document]
Comment (s3alfisc) [reply docs/quickstart.ipynb:34]: Solved with [a86f14d](https://github.com/py-econometrics/pyfixest/pull/569/commits/a86f14d98c426c45b09c901f66ec07dd8c7b36d1) =)

Comment (s3alfisc) [reply docs/quickstart.ipynb:34]: Added with [fa5cbec](https://github.com/py-econometrics/pyfixest/pull/569/commits/fa5cbec20c7ab1e263a70b63ce6eab25d622d9b8)
